apps/api/src/server/applications/subscriptions/subscriptions.service.js

```diff
@@ -0,0 +1,80 @@
+import { eventClient } from '../../infrastructure/event-client';
+import { NSIP_SUBSCRIPTION } from '../../infrastructure/topics';
+import * as subscriptionRepository from '../../repositories/subscription.respository.js';
+import {
+	buildSubscriptionPayloads,
+	subscriptionTypeChanges,
+	typesToSubscription
+} from './subscriptions';
+import { EventType } from '@pins/event-client';
+
+/**
+ * Create or update a subscription, and publishes the corresponding event.
+ *
+ * @param {any} request
+ * @returns {Promise<{id: number, created: boolean}>}
+ */
+export async function createOrUpdateSubscription(request) {
+	const subscription = prepareInput(request);
+
+	const existing = await subscriptionRepository.findUnique(
```

Comment:
Just an observation but if caseRef or email is passed in uppercase and lowercase we will end up with 2 subscriptions.  I'm not really proposing we enforce caps though, just something to be aware of.

Reply:
Good catch, I would normally enforce lowercase for emails - since in practice they are case insensitive. 

Given our (back-office) API is just internal, I think we can get away with it for now!